APMrover2/sensors.cpp

```diff
@@ -89,7 +89,10 @@ void Rover::update_wheel_encoder()
          * timeStamp_ms is the time when the rotation was last measured (msec)
          * posOffset is the XYZ body frame position of the wheel hub (m)
          */
-        EKF3.writeWheelOdom(delta_angle, delta_time, wheel_encoder_last_update_ms[i], g2.wheel_encoder.get_pos_offset(i), g2.wheel_encoder.get_wheel_radius(i));
+            // Update EKF3 wheel odometry buffer only if EKF3 is used
+        if (ahrs.get_ekf_type() == 3) {
```

Comment:
We want to supply this data to the EKF even if it isn't the active EKF - that way we can see its outputs but it can't crash the vehicle with bad outputs.